api/providers/ContactProvider.cs

```diff
@@ -46,4 +47,23 @@ public Contact Delete(Guid id)
     {
         throw new NotImplementedException();
     }
+
+    public Contact? GetOrCreateContact(SupplierRequest request)
+    {
+
+        if (request.Contact_id != null)
+        {
+            Contact? existingContact = GetById(request.Contact_id.Value);
```

Comment:
```suggestion
            Contact? existingContact = GetById(request.ContactId.Value);
```

Graag PascalCase aanhouden. (en vranderen in de ContactDTO)